File: 5th/Simple_Linear_Regression.jl
A = readcsv("data.csv", header=true)

X = A[1][:,2]
Y = A[1][:,3]

function beta1(X,Y)
  n = 10
  b = (n*sum(X.*Y) - (sum(X)*sum(Y)))/(n*sum(X.^2) - sum(X)^2)
  return b
end

function beta0(X,Y)
  n = 10
  b0 = (sum(Y) - beta1(X,Y)*sum(X))/n
  return b0
end

hatY = beta0(X,Y) + beta1(X,Y)*X

B = hcat(A[1], hatY)

writecsv("slr.csv", B)

File: 5th/Simple_Linear_Regression.R
library("ggplot2")

READ <- read.csv("slr.csv", header = F)

X <- READ["V2"]
Y <- READ["V3"]
hatY <- READ["V4"]

ggplot(READ)+ geom_line(aes(V2, V4, color="reg")) + geom_point(aes(V2, V3, color="data")) + labs(x = "X", y = "Y")
